### PR TITLE
:warning: Update Kubernetes client libraries to 1.15.0 release

### PR DESCRIPTION
**Pull request by munnerz**
This is #231 by @tamalsaha rebased onto master.

I'm not 100% sure how dependency files are meant to be managed in this repo as there's no `hack/` directory, and the repo also includes both Gopkg.* files as well as go.{mod,sum}.

/cc @DirectXMan12

### CODE REVIEW

**Comment by munnerz** [comment]
I've tried to reproduce the test failures locally but don't seem to be able to. Maybe caused by a network flake? I've force pushed without changing anything to find out 😄

**Comment by munnerz** [comment]
🎉 ✨  increasing the default golintci timeout slightly seems to have got the tests passing! This should be good to go 😄 

/assign @DirectXMan12

**Comment by munnerz** [comment]
@DirectXMan12 I've rebased this PR - anything I can do to nudge this along, or any reason it's being held? 😄

**Comment by DirectXMan12** [comment]
oh, shoot.  Lost track of this one.

File: pkg/webhook/parser.go [hunk 1]
```diff
@@ -141,7 +146,61 @@ func (c Config) ToWebhook() admissionreg.Webhook {
...
+		failurePolicy = admissionreg.Fail
+	default:
+		failurePolicy = admissionreg.FailurePolicyType(c.FailurePolicy)
+	}
+	path := c.Path
+	return admissionreg.ValidatingWebhook{
+		Name:          c.Name,
+		Rules:         []admissionreg.RuleWithOperations{whConfig},
+		FailurePolicy: &failurePolicy,
+		ClientConfig: admissionreg.WebhookClientConfig{
```
Comment (DirectXMan12) [inline pkg/webhook/parser.go]: can we refactor the more complicated bits out into methods, so it'll be harder to accidentally change on but not the other?

Comment (mengqiy) [reply pkg/webhook/parser.go]: +1
Any update?

Comment (marun) [reply pkg/webhook/parser.go]: @DirectXMan12 #318 is a refactor that is hopefully in keeping with you request.

Comment (DirectXMan12) [reply pkg/webhook/parser.go]: cool, thanks!

**Comment by DirectXMan12** [comment]
rebase so we can get this merged.

**Pull request merged**